src/test/java/org/broadinstitute/hellbender/tools/spark/CreateIndexForHadoopBamIntegrationTest.java

```diff
@@ -0,0 +1,132 @@
+    public void testNegativeGranularity(){
+        ArgumentsBuilder args = getInputAndOutputArgs(SORTED_BAM, getTempIndexFile())
+            .add("--"+ CreateIndexForHadoopBam.SPLITTING_INDEX_GRANULARITY_LONG_NAME).add(-10);
+        this.runCommandLine(args);
+    }
+
+    @DataProvider(name="unindexable")
+    public Object[][] getUnindexableFiles(){
+        return new Object[][]{
+                {getTestFile("count_reads.cram")},
+                {getTestFile("count_reads.sam")}
+        };
+    }
+
+    @Test(expectedExceptions = UserException.BadInput.class, dataProvider = "unindexable")
+    public void testUnindexableFilesFail(File badFile){
+        ArgumentsBuilder args = getInputAndOutputArgs(badFile, getTempIndexFile());
+        this.runCommandLine(args);
+    }
+
+    @Test(dataProvider = "indexable")
+    public void testUnspecifiedOutputProducesAdjacentIndex(File bam) throws IOException {
+        File bamCopy = createTempFile("copy-"+bam, ".bam");
+        Files.copy(bam.toPath(), bamCopy.toPath(), StandardCopyOption.REPLACE_EXISTING);
+        File expectedIndex = new File(bamCopy.toPath() + SplittingBAMIndexer.OUTPUT_FILE_EXTENSION);
+        Assert.assertFalse(expectedIndex.exists());
+        ArgumentsBuilder args = ArgumentsBuilder.init().addInput(bamCopy);
+        this.runCommandLine(args);
+        expectedIndex.deleteOnExit();
+        assertIndexIsNotEmpty(expectedIndex);
+    }
+
+    @Test
+    public void testBothPathsProduceSameIndex() throws IOException {
+        final File splittingIndexOnly = getTempIndexFile();
+        final ArgumentsBuilder splittingIndexOnlyArgs = getInputAndOutputArgs(SORTED_BAM, splittingIndexOnly);
+        this.runCommandLine(splittingIndexOnlyArgs);
+        assertIndexIsNotEmpty(splittingIndexOnly);
+
+        final File splittingIndexWithBai = getTempIndexFile();
+        final ArgumentsBuilder splittingAndBaiArgs = getInputAndOutputArgs(SORTED_BAM, splittingIndexWithBai)
+                .add("--"+ CreateIndexForHadoopBam.CREATE_BAI_LONG_NAME);
+        this.runCommandLine(splittingAndBaiArgs);
+        assertIndexIsNotEmpty(splittingIndexWithBai);
+
+        IOUtil.assertFilesEqual(splittingIndexOnly, splittingIndexWithBai);
+    }
+
+    public ArgumentsBuilder getInputAndOutputArgs(File inputFile, File splittingIndexWithBai) {
+        return ArgumentsBuilder.init()
+                .addInput(inputFile)
+                .addOutput(splittingIndexWithBai);
+    }
+
+    @Test
+    public void testCreateWithBaiCreatesBai(){
+        final File splittingIndex = getTempIndexFile();
+        final File baiIndex = IOUtils.replaceExtension(splittingIndex, BAMIndex.BAMIndexSuffix);
+        Assert.assertFalse(baiIndex.exists());
+        final ArgumentsBuilder args = getInputAndOutputArgs(SORTED_BAM, splittingIndex)
+                .add("--" + CreateIndexForHadoopBam.CREATE_BAI_LONG_NAME);
+        this.runCommandLine(args);
+        Assert.assertTrue(baiIndex.exists());
```

Comment:
will the splitting bam file be removed at the end of this test? BaseTest.createTempFile does only handles `bai` I think